packages/ra-core/src/controller/input/ReferenceInputController.tsx

```diff
@@ -16,9 +15,56 @@ import {
     getReferenceResource,
 } from '../../reducer';
 import { getStatusForInput as getDataStatus } from './referenceDataStatus';
-import translate from '../../i18n/translate';
+import withTranslate from '../../i18n/translate';
+import { Sort, Translate, Record, Pagination, Dispatch } from '../../types';
+import { MatchingReferencesError } from './types';
 
-const referenceSource = (resource, source) => `${resource}@${source}`;
+const defaultReferenceSource = (resource: string, source: string) =>
+    `${resource}@${source}`;
+
+interface ChildrenFuncParams {
+    choices: Record[];
+    error?: string;
+    filter?: any;
+    isLoading: boolean;
+    onChange: (value: any) => void;
+    pagination: Pagination;
+    setFilter: (filter: any) => void;
+    setPagination: (pagination: Pagination) => void;
+    setSort: (sort: Sort) => void;
+    sort: Sort;
+    warning?: string;
+}
+
+interface Props {
+    allowEmpty?: boolean;
+    basePath: string;
+    children: (params: ChildrenFuncParams) => ReactNode;
+    crudGetMatchingAccumulate: Dispatch<typeof crudGetMatchingAccumulateAction>;
+    crudGetManyAccumulate: Dispatch<typeof crudGetManyAccumulateAction>;
+    filter?: object;
+    filterToQuery: (filter: {}) => any;
+    input?: {
```

Comment:
don't we have a type from redux-form for that?

Reply:
We would have to include the redux-form types in our dev deps and last time I tried that, they were not good

Reply:
I added them and it seems ok